ompi/mpi/c/init.c.in

```diff
@@ -46,7 +47,8 @@ PROTOTYPE INT init(INT_OUT argc, ARGV argv)
 
     if (NULL != (env = getenv("OMPI_MPI_THREAD_LEVEL"))) {
         required = atoi(env);
-        if (required < MPI_THREAD_SINGLE || required > MPI_THREAD_MULTIPLE) {
+        if (required != MPI_THREAD_SINGLE && required != MPI_THREAD_FUNNELED &&
+            required != MPI_THREAD_SERIALIZED && required != MPI_THREAD_MULTIPLE) {
```

Comment:
Just curious: what's the need for this?  Doesn't MPI specify the relationship between all the `MPI_THREAD_*` values such that < SINGLE and > MULTIPLE is guaranteed to be invalid?

Reply:
In the new MPI ABI, values are still ordered but not consecutive 

https://github.com/mpi-forum/mpi-abi-stubs/blob/main/mpi.h

Reply:
Actually, this is worth a comment for those of us who look at this code and wonder why it is this way (and to make sure someone doesn't mistakenly return it to <SINGLE || >MULTIPLE someday).  Can you please add comments in these places and describe that checking each individual value is required because of ABI?  Thanks.